src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -140,7 +144,55 @@ bool QgsPointCloudLayerRenderer::render()
     mAttributes.push_back( mLayerAttributes.at( layerIndex ) );
   }
 
-  QgsPointCloudDataBounds db;
+  const QgsRectangle renderExtent = renderContext()->coordinateTransform().transformBoundingBox( renderContext()->mapExtent(), Qgis::TransformDirection::Reverse );
+  bool canceled = false;
+  if ( !hasMultipleIndexes )
+  {
+    canceled = !renderIndex( pc );
+  }
+  else
+  {
+    const auto subLayers = mLayer->dataProvider()->subIndexes();
```

Comment:
It's not thread safe to access a layer here -- you need to copy everything you need upfront in the constructor.